paltas/Analysis/dataset_generation.py

```diff
@@ -15,7 +15,7 @@
 from lenstronomy.SimulationAPI.observation_api import SingleBand
 import warnings
 from scipy.ndimage import rotate
-
+import h5py
```

Comment:
Is h5py already installed by one of our dependencies? If not, you could do e.g.
```python
try:
    import h5py
except ImportError:
    h5py = None
```
and maybe print a useful message if someone tries to use  `--h5` without having h5py installed. Or we could just add h5py to our requirements.txt -- if you end up writing a unit test for this, that would be the easiest move.